go/controller/api/v1alpha1/deploymentsettings_types.go

```diff
@@ -176,6 +180,50 @@ func (r *ElasticsearchConnection) Attributes(ctx context.Context, c client.Clien
 	return attr, nil
 }
 
+type OpensearchConnection struct {
+	// Host ...
+	//
+	// +kubebuilder:validation:Required
+	Host string `json:"host"`
+
+	// Index to query in opensearch
+	//
+	// +kubebuilder:validation:Optional
+	Index string `json:"index"`
+
+	// AWS Access Key ID to use, can also use IRSA to acquire credentials
+	//
+	// +kubebuilder:validation:Optional
+	AWSAccessKeyID *string `json:"awsAccessKeyId,omitempty"`
+
+	// AWS Secret Access Key to use, can also use IRSA to acquire credentials
+	//
+	// +kubebuilder:validation:Optional
+	AwsSecretAccessKeyRef *corev1.SecretKeySelector `json:"awsSecretAccessKeyRef,omitempty"`
```

Comment:
I'd probably name this `awsSecretAccessKeySecretRef` actually (which is clunky but more accurate)